File: Kotlin/fizzBuzzRecursive.kt
// A simple Recursive kotlin program for fizzbuzz
// Author: @JossJoestar

import java.util.Scanner

fun main(args:Array<String>){ 
    print("Give me a number please: ")
    val read = Scanner(System.`in`)  
    var number = read.nextInt()  
    fizz(1,number)
}

fun fizz(n:Int, end:Int){
    if(n%3 == 0 && n%5 ==0)
        println("FizzBuzz")
    else if(n%5==0)
        println("Buzz")
    else if(n%3==0)
        println("Fizz")
    else
        println("") 
    if(n < end)
        return fizz(n+1, end)
}

File: Go/fizzbuzz-recursive.go
// A recursive implementation of FizzBuzz in Go
// Author: @evamy

package main

import (
	"fmt"
	"strconv"
)

func main() {
	PrintFizzBuzz(100)
}

func PrintFizzBuzz(n int) {
	fizzbuzz(1, n)
}

func fizzbuzz(current int, n int) {
	if (current%3 == 0 && current%5 == 0) {
		fmt.Println("FizzBuzz")
	} else if current%3 == 0 {
		fmt.Println("Fizz")
	} else if current%5 == 0 {
		fmt.Println("Buzz")
	} else {
		fmt.Println(strconv.Itoa(current))
	}

	if current < n {
		fizzbuzz(current+1, n)
	}

	return
}